crates/walrus-service/src/node/shard_sync.rs

```diff
@@ -72,16 +72,19 @@ impl ShardSyncHandler {
     ) -> Result<(), SyncShardClientError> {
         let mut task_handle = self.task_handle.lock().await;
         let sync_handler = self.clone();
+
+        // If there is an existing task, we need to abort it first before starting a new one.
+        if let Some(old_task) = task_handle.take() {
+            old_task.abort();
+        }
+
         let new_task = tokio::spawn(async move {
             sync_handler
                 .sync_shards_task(shards, recover_metadata)
                 .await
         });
 
-        // Abort any existing task before replacing it
-        if let Some(old_task) = task_handle.replace(new_task) {
-            old_task.abort();
-        }
+        *task_handle = Some(new_task);
```

Comment:
Question: Is this change actually necessary or does it just make reasoning about it easier?